Eastern-Research-Group/mywaterway
# HMW-664 Switched from react-window to react-virtuoso

## Related Issues:
* [HMW-664](https://jira.epa.gov/browse/HMW-659)

## Main Changes:
* Switched from using the `react-window` package to [`react-virtuoso`](https://github.com/petyosi/react-virtuoso) for smoother list virtualization. The `react-virtuoso` package is smaller, it uses `ResizeObserver` out of the box (for dynamic content like the CyAN content), and otherwise requires no imperative item sizing.

## Steps To Test:
1. Go to http://192.168.0.244:3000/community/lake%20mattamuskeet/overview, and click on _Water Monitoring Locations_.
2. With all monitoring types selected, scroll down to the CyAN waterbody. The list should scroll smoothly along the way.
3. Expand the CyAN content, and wait for the content to load. The accordion item should resize to fit the content.
4. Test the behavior of the other virtualized list instances, i.e. the characteristics filter, the waterbody list, and the list of waters on the Plan Summary page.
